Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,829 @@
+                if ($logControl.ControlId -eq $controlId) {
+                    $logControl.SkippedResources = $logSkippedResources
+                }
+            }
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$( $totalDBForMySQLFSWithSSLNotEnabled )] Azure Database for MySQL flexible server(s) for which SSL is disabled." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)	
+
+    $colsProperty = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.SSLStatus }; Label = "'SSL Status'"; Width = 30; Alignment = "Center" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" }
+
+    $colsProperty1 = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" }
+
+    if (-not $AutoRemediation) {
+        Write-Host "Azure Database for MySQL flexible server(s) with SSL disabled are:"
+         $DBForMySQLFSWithSSLNotEnabled  | Format-Table -Property $colsProperty1 -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }    
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetSSLForDBForMySQLFlexibleServer"
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "[Step 3 of 4] Back up Azure Database for MySQL flexible server(s) details."
+    Write-Host $([Constants]::SingleDashLine)
+
+    if ([String]::IsNullOrWhiteSpace($FilePath)) {
+        # Backing up Azure Database for MySQL flexible server(s) details.
+        $backupFile = "$($backupFolderPath)\DBForMySQLFlexibleServerWithSSLDisabled.csv"
+         $DBForMySQLFSWithSSLNotEnabled  | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "Azure Database for MySQL flexible server(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else {
+        Write-Host "Skipped as -FilePath is provided." -ForegroundColor $([Constants]::MessageType.Warning)
+    }
+
+    if (-not $DryRun) {
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "[Step 4 of 4] Enable SSL for Azure Database for MySQL flexible server(s) in the Subscription." 
```

Comment:
If you will take this statement out of this if block, you don't need to write it again the else block and it will again simplify the code.